models/sudoswap/ethereum/sudoswap_ethereum_pool_settings_latest.sql

```diff
@@ -0,0 +1,168 @@
+                ,newSpotPrice/1e18 as spot_price
+                ,evt_block_time as update_time
+                ,row_number() over (partition by contract_address order by evt_block_number desc, tx.index desc) as ordering
+            FROM {{ source('sudo_amm_ethereum','LSSVMPair_general_evt_SpotPriceUpdate') }} evt
+            INNER JOIN {{ source('ethereum','transactions') }} tx ON tx.hash = evt.evt_tx_hash
+            {% if not is_incremental() %}
+            AND tx.block_time >= '{{project_start_date}}'
+            AND evt.evt_block_time >= '{{project_start_date}}'
+            {% endif %}
+            {% if is_incremental() %}
+            AND tx.block_time >= date_trunc("day", now() - interval '1 week')
+            AND evt.evt_block_time >= date_trunc("day", now() - interval '1 week')
+            {% endif %}
+        ) foo
+    WHERE ordering = 1
+)
+
+, latest_settings as (
+    select
+        coalesce(t1.pool_address, t2.pool_address, t3.pool_address) as pool_address
+        ,pool_fee
+        ,delta
+        ,spot_price
+        ,coalesce(t1.update_time, t2.update_time, t3.update_time) as latest_update_time
+    from latest_spot_price t1
+    full join latest_delta t2 on t1.pool_address = t2.pool_address
+    full join latest_pool_fee t3 on t1.pool_address = t3.pool_address or t2.pool_address = t3.pool_address
+)
+
+, initial_settings as (
+    SELECT
+      output_pair AS pool_address,
+      CASE
+        WHEN _bondingCurve = '{{linear_bonding_address}}' THEN 'linear'
+        WHEN _bondingCurve = '{{exponential_bonding_address}}' THEN 'exponential'
+        ELSE 'other'
+      END as bonding_curve,
+      _spotPrice / 1e18 AS spot_price,
+      _delta / 1e18 as delta,
+      _fee AS pool_fee
+      ,call_block_time as creation_block_time
+    FROM
+      {{ source('sudo_amm_ethereum','LSSVMPairFactory_call_createPairETH') }}
+    WHERE call_success
+    {% if is_incremental() %}
+    AND call_block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+)
+
+-- incremental update:
+-- we need to backfill columns from the existing data in order to have full rows
+{% if is_incremental() %}
+, full_settings_backfilled as (
+    select * from(
+        select
+         coalesce(t1.pool_address,t3.pool_address) as pool_address
+        ,coalesce(t2.bonding_curve, t3.bonding_curve) as bonding_curve
+        ,coalesce(t1.pool_fee, t2.pool_fee, t3.pool_fee) as pool_fee
+        ,coalesce(t1.delta, t2.delta, t3.delta) as delta
+        ,coalesce(t1.spot_price, t2.spot_price, t3.spot_price) as spot_price
+        ,coalesce(t1.latest_update_time,t3.creation_block_time) as latest_update_time
+        from latest_settings t1
+        full outer join initial_settings t3
+            ON t1.pool_address = t3.pool_address
+        left join {{ this }} t2
+            ON t1.pool_address = t2.pool_address
+    ) foo
+    where bonding_curve is not null --temp hack to exclude updates form erc20 pools
+)
+{% endif %}
```

Comment:
@jeff-dude please take a closer look at this setup as it uses the existing table to fill empty columns on an incremental update.